src/network/networkd-dhcp4.c

```diff
@@ -1256,8 +1293,12 @@ static int dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                         }
                         if (r > 0)
                                 return -ENOMSG;
-                        break;
 
+                        /* IPv6-Only Preferred Option for DHCPv4 */
+                        if (link->network->dhcp_allow_ipv6_only_mode && dhcp_offer_contains_ipv6_only_preferred_option(link, client))
```

Comment:
No no, in this stage, the client already sent REQUEST and received ACK.
The RFC states that the client should start the `V6ONLY_WAIT` timer when received OFFER, and the client should not send REQUEST in that case.
https://datatracker.ietf.org/doc/html/rfc8925#name-dhcpv4-client-behavior
> If the client includes the IPv6-Only Preferred option code in the Parameter Request List and the DHCPOFFER message from the server contains a valid IPv6-Only Preferred option, the client SHOULD NOT request the IPv4 address provided in the DHCPOFFER. If the IPv6-Only Preferred option returned by the server contains a value greater than or equal to MIN_V6ONLY_WAIT, the client SHOULD set the V6ONLY_WAIT timer to that value. Otherwise, the client SHOULD set the V6ONLY_WAIT timer to MIN_V6ONLY_WAIT. The client SHOULD stop the DHCPv4 configuration process for V6ONLY_WAIT seconds or until a network attachment event, whichever happens first.

Reply:
This comes in after getting the OFFER from server . ```client_handle_offer``` . We not proceeding to request

1. Discover: 
2. Offer: <HERE we are calling after receiving the offer> 
3.  Request:

Reply:
Ah, indeed. Sorry. I forgot that the handler is also called after OFFER.

But, still that should not be handled here, but in sd-dhcp-client.c.
There is nothing we need to change in this handler.
Only we need to do in networkd side is stopping the DHCPv4 client when NDisc or DHCPv6 configures an IPv6 address.

Reply:
The flag ```dhcp_allow_ipv6_only_mode``` which we get from user. On basic of that this operation happens. It's similar to we filter the black/white listed dhcp servers.